apps/api/src/http/internal/rest/webhooks/stripe/subscription.rs

```diff
@@ -0,0 +1,308 @@
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use mongodb::options::{FindOneAndUpdateOptions, ReturnDocument, UpdateOptions};
+use shared::database::product::subscription::{
+	ProviderSubscriptionId, Subscription, SubscriptionId, SubscriptionPeriod, SubscriptionState,
+};
+use shared::database::product::{ProductId, SubscriptionProduct, SubscriptionProductVariant};
+use shared::database::queries::{filter, update};
+use shared::database::user::UserId;
+
+use crate::global::Global;
+use crate::http::error::ApiError;
+use crate::transactions::{TransactionError, TransactionResult, TransactionSession};
+
+fn subscription_products(items: stripe::List<stripe::SubscriptionItem>) -> Result<Vec<ProductId>, ApiError> {
+	items
+		.data
+		.into_iter()
+		.map(|i| Ok(ProductId::from(i.price.ok_or(ApiError::BAD_REQUEST)?.id)))
+		.collect()
+}
+
+/// Creates the subscription object in the database.
+pub async fn created(
+	_global: &Arc<Global>,
+	mut tx: TransactionSession<'_, ApiError>,
+	subscription: stripe::Subscription,
+) -> TransactionResult<(), ApiError> {
+	let items = subscription_products(subscription.items).map_err(TransactionError::custom)?;
+
+	let products = tx
+		.find(
+			filter::filter! {
+				SubscriptionProduct {
+					#[query(flatten)]
+					variants: SubscriptionProductVariant {
+						#[query(selector = "in", serde)]
+						id: items,
+					}
+				}
+			},
+			None,
+		)
+		.await?;
+
+	if products.len() != 1 {
+		// only accept subs with one product
+		return Ok(());
+	}
+
+	let user_id = subscription
+		.metadata
+		.get("USER_ID")
+		.and_then(|i| UserId::from_str(i).ok())
+		.ok_or(TransactionError::custom(ApiError::BAD_REQUEST))?;
+
+	let product = products.into_iter().next().unwrap();
+
+	let sub_id = SubscriptionId {
+		user_id,
+		product_id: product.id,
+	};
+
+	tx.update_one(
+		filter::filter! {
+			Subscription {
+				#[query(rename = "_id", serde)]
+				id: &sub_id,
+			}
+		},
+		update::update! {
+			#[query(set_on_insert)]
+			Subscription {
+				id: sub_id.clone(),
+				state: SubscriptionState::Active,
+				updated_at: chrono::Utc::now(),
+			}
+		},
```

Comment:
Is this correct? If they were previously subbed this operation wouldnt do anything?
Are they subbed if we get a create event or do we need to wait for another event? 
What period is this for?